src/Modal/ModalContent.tsx

```diff
@@ -0,0 +1,195 @@
+    e.preventDefault();
+
+    const startDateTime = new Date(`${startDate}T${startTime}`);
+    const endDateTime = new Date(`${endDate}T${endTime}`);
+
+    if (endDateTime < startDateTime) {
+      setErrorMessage(
+        '* End date and time cant be earlier than start date and time.'
+      );
+      return;
+    }
+    const event: Event = {
+      id: generateSimpleID(),
+      eventTitle,
+      startDateTime,
+      endDateTime,
+      description,
+    };
+
+    onSave(event);
+  };
+
+  return (
+    <div className={styles.modalContent}>
+      <CloseButton onClose={onClose}></CloseButton>
+      <h2>Create New Event</h2>
+      <form className={styles.addEventForm} onSubmit={handleSubmit}>
+        <LabelInputPair
+          inputClassName={styles.eventFormInput}
+          parentDivClassName={styles.datetime}
+          labelText="Event Title:"
+          inputType="text"
+          inputName="eventTitle"
+          value={eventTitle}
+          onChange={setEventTitle}
+        />
+        <div className={styles.dateTimeGroup}>
+          <LabelInputPair
+            inputClassName={styles.eventFormInput}
+            parentDivClassName={styles.datetime}
+            labelText="Start Date:"
+            inputType="date"
+            inputName="startDate"
+            value={startDate}
+            onChange={setStartDate}
+          />
+
+          <LabelInputPair
+            inputClassName={styles.eventFormInput}
+            parentDivClassName={styles.datetime}
+            labelText="Start Time:"
+            inputType="time"
+            inputName="startTime"
+            value={startTime}
+            onChange={setStartTime}
+          />
+        </div>
+        <div className={styles.dateTimeGroup}>
+          <LabelInputPair
+            inputClassName={styles.eventFormInput}
+            parentDivClassName={styles.datetime}
+            labelText="End Date"
+            inputType="date"
+            inputName="endDate"
+            value={endDate}
+            onChange={setEndDate}
+          />
+
+          <LabelInputPair
+            inputClassName={styles.eventFormInput}
+            parentDivClassName={styles.datetime}
+            labelText="End Time:"
+            inputType="time"
+            inputName="endTime"
+            value={endTime}
+            onChange={setEndTime}
+          />
+        </div>
+        <span className={styles.errorMessage}>{errorMessage}</span>
+
+        <LabelTextAreaPair
+          labelText="Description:"
+          textAreaName="eventDescription"
+          value={description}
+          onChange={setDescription}
+          textAreaClassName={styles.eventFormInput}
+        />
+        <CreateEventButton />
+      </form>
+    </div>
+  );
+};
+export const CloseButton = ({ onClose }: CloseButtonProps) => {
+  return (
+    <span className={styles.close} onClick={onClose}>
+      &times;
+    </span>
+  );
+};
+
+export const LabelInputPair = ({
```

Comment:
I think these components are too big to be at the bottom, the file is too long, so they can be moved to individual components